WS17_18/02_mvc/your assignment/hahn/Yatodoa/Yatodoa/view/TodoView.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  TodoView.swift
+//  Done
+//
+//  Created by Alex on 07.11.17.
+//  Copyright © 2017 Alexander Dobrynin. All rights reserved.
+//
+
+import UIKit
+
+protocol TodoViewDelegate {
+    func todoViewHasChanged(task: String, completed: Bool, stamp: Date)
+}
+
+class TodoView: CustomView {
+
+    @IBOutlet weak var checkbox: TDCheckbox!
+    @IBOutlet weak var textField: UITextField!
+    @IBOutlet weak var label: UILabel!
+    
+    var delegate: TodoViewDelegate?
+    //var parentController: TodoViewController? // FIXME: replace with delegate pattern
+    
+    @IBAction func buttonChecked(_ sender: TDCheckbox) {
+        if let todo = textField.text, !todo.isEmpty {
+            let newValue = checkbox.check()
+            let now = Date()
+            
+            if newValue {
+                textField.attributedText = NSAttributedString(string: todo, attributes: [
+                    NSAttributedStringKey.foregroundColor: UIColor.lightGray,
+                    NSAttributedStringKey.strikethroughStyle: NSUnderlineStyle.styleSingle.rawValue
+                    ])
+                //textField.alpha = 0.5
+                let dateFormatter = DateFormatter()
+                dateFormatter.locale = Locale(identifier: "de_DE")
```

Comment:
Zur Info: Der `DateFormatter` hat die Locale des Devices als Standard-Local gesetzt. In den meisten Fällen muss man das nicht angeben. Im Simulator ist das aber berechtigt, weil der Simulator standardmäßig auf US gesetzt ist